File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_61786.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request


async def BenchmarkTest61786(request: Request):
    forwarded_ip = request.headers.get('x-forwarded-for', '')
    data = f'{forwarded_ip:.200s}'
    data = bytearray(int(data) if str(data).isdigit() else 0)
    return {"updated": True}

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest61786.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest61786 {

    @PostMapping(path="/BenchmarkTest61786", consumes="application/xml")
    public void BenchmarkTest61786(@RequestBody String xmlBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String xmlValue = xmlBody;
        String data = xmlValue.replace("\u0000", "");
        String routeResult = "unknown";
        switch (data) {
            case "retry": routeResult = "retry-handled"; break;
            case "abort": routeResult = "abort-handled"; break;
        }
        response.setHeader("X-Route-Result", routeResult);
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}
